components/ILIAS/Forum/classes/Moderation/class.ForumModeratorsTable.php

```diff
@@ -0,0 +1,198 @@
+<?php
+
+/**
+ * This file is part of ILIAS, a powerful learning management system
+ * published by ILIAS open source e-Learning e.V.
+ *
+ * ILIAS is licensed with the GPL-3.0,
+ * see https://www.gnu.org/licenses/gpl-3.0.en.html
+ * You should have received a copy of said license along with the
+ * source code, too.
+ *
+ * If this is not the case or you just want to try ILIAS, you'll find
+ * us at:
+ * https://www.ilias.de
+ * https://github.com/ILIAS-eLearning
+ *
+ *********************************************************************/
+
+declare(strict_types=1);
+
+use ILIAS\Data;
+use ILIAS\UI;
+use Psr\Http\Message\ServerRequestInterface;
+
+class ForumModeratorsTable
+{
+    private ilForumModerators $forum_moderators;
+    protected \ilCtrl $ctrl;
+    protected \ilLanguage $lng;
+    protected UI\Factory $ui_factory;
+    protected ServerRequestInterface $request;
+    protected Data\Factory $data_factory;
+
+    public function __construct(ilForumModerators $forum_moderators)
+    {
+        global $DIC;
+
+        $this->ctrl = $DIC->ctrl();
+        $this->lng = $DIC->language();
+        $this->ui_factory = $DIC->ui()->factory();
+        $this->request = $DIC->http()->request();
+        $this->data_factory = new Data\Factory();
+        $this->forum_moderators = $forum_moderators;
+    }
+
+    public function getComponent(): UI\Component\Table\Data
+    {
+        $columns = $this->getColumns();
+        $actions = $this->getActions();
+        $data_retrieval = $this->getDataRetrieval();
+
+        return $this->ui_factory->table()
+                                ->data($this->lng->txt('frm_moderators'), $columns, $data_retrieval)
+                                ->withActions($actions)
+                                ->withRequest($this->request);
+    }
+
+    protected function getColumns(): array
+    {
+        return [
+            'usr_id' => $this->ui_factory->table()->column()->number('User ID')
+                                         ->withIsSortable(true),
+
+            'login' => $this->ui_factory->table()->column()->text($this->lng->txt('login'))
+                                        ->withIsSortable(true),
+
+            'firstname' => $this->ui_factory->table()->column()->text($this->lng->txt('firstname'))
+                                            ->withIsSortable(true),
+
+            'lastname' => $this->ui_factory->table()->column()->text($this->lng->txt('lastname'))
+                                           ->withIsSortable(true),
+        ];
+    }
+
+    protected function getActions(): array
+    {
+        $query_params_namespace = ['frm_moderators_table'];
+
+        $uri_detach = $this->data_factory->uri(
```

Comment:
And another issue, sorry.

Please change the implementation to the one I used for `SAML`. There should be an action handling method with a switch/match statement to delegate the action to the corresponding method (even if it is only `ban-delete` here) in the parent class (here: ilForumModeratorsGUI::class).